### PR TITLE
Split Ride and rct2_ride

### PR DESCRIPTION
**Pull request by Gymnasiast**
This is another step in moving away from SV6.

### CODE REVIEW

File: src/openrct2/ride/ride_ratings.h [hunk 1]
```diff
@@ -18,7 +18,23 @@
...
-#include "ride.h"
+
+typedef fixed16_2dp ride_rating;
+
+// Convenience function for writing ride ratings. The result is a 16 bit signed
+// integer. To create the ride rating 3.65 type RIDE_RATING(3,65)
+#define RIDE_RATING(whole, fraction)    FIXED_2DP(whole, fraction)
+#define RIDE_RATING_UNDEFINED           (ride_rating)(uint16)0xFFFF
+
+#pragma pack(push, 1)
```
Comment (janisozaur) [inline src/openrct2/ride/ride_ratings.h]: here

**Comment by janisozaur** [comment]
If `Ride` and `rct2_ride` are really meant to be split, `Ride` should be have field-wise assignments and be moved out of packing pragmas

**Comment by Gymnasiast** [comment]
You mean doing `ride->property = 123`, rather than memcpy'ing?

**Comment by janisozaur** [comment]
yes, you already did that for sv6, right?

**Comment by Gymnasiast** [comment]
Yes, the exporter is not the problem here. But the NO_RCT2 stuff is. Unless gRideList is not needed for TestPaint.

**Comment by Gymnasiast** [comment]
Ok, changed the S6Exporter. But I also checked TestPaint, and it does need gRideList, so `Ride` will have to stay `pack`ed for now.

**Comment by janisozaur** [comment]
xx

Please don't do that.

**Comment by Gymnasiast** [comment]
It's Github that does that.

**Comment by janisozaur** [comment]
No, it's you putting more than 70 chars in subject line of a commit.

**Comment by Gymnasiast** [comment]
And? I'm not going to create non-descriptive comments just because Github fucks them up.

**Comment by Gymnasiast** [comment]
I have changed the commit message and added a comment about Testpaint.

**Pull request merged**